src/main/java/com/devonfw/demoquarkus/logic/UcFindAnimalImpl.java

```diff
@@ -0,0 +1,93 @@
+package com.devonfw.demoquarkus.logic;
+
+import com.devonfw.demoquarkus.domain.model.AnimalEntity;
+import com.devonfw.demoquarkus.domain.repo.AnimalRepository;
+import com.devonfw.demoquarkus.service.v1.mapper.AnimalMapper;
+import com.devonfw.demoquarkus.service.v1.model.AnimalDto;
+import com.devonfw.demoquarkus.service.v1.model.AnimalSearchCriteriaDto;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+
+@Named
+@Transactional
+@Slf4j
+public class UcFindAnimalImpl implements UcFindAnimal {
+    @Inject
+    AnimalRepository animalRepository;
+
+    @Inject
+    AnimalMapper mapper;
+
+    @Override
+    public Page<AnimalDto> findAnimals(AnimalSearchCriteriaDto dto) {
+        Iterable<AnimalEntity> animalsIterator = this.animalRepository.findAll();
+        List<AnimalEntity> animals = new ArrayList<AnimalEntity>();
+        animalsIterator.forEach(animals::add);
+        List<AnimalDto> animalsDto = this.mapper.map(animals);
+        return new PageImpl<>(animalsDto, PageRequest.of(dto.getPageNumber(), dto.getPageSize()), animalsDto.size());
+    }
+
+    @Override
+    public Page<AnimalDto> findAnimalsByCriteriaApi(AnimalSearchCriteriaDto dto) {
+        List<AnimalEntity> animals = this.animalRepository.findAllCriteriaApi(dto).getContent();
```

Comment:
Fine here...
https://github.com/devonfw-sample/devon4quarkus-reference/blob/master/src/main/java/com/devonfw/demoquarkus/domain/repo/AnimalFragmentImpl.java#L50

Later we can improve the code and extract some helper classes for such boiler-plate code to:
* do the paging in fragment (as this code pattern will otherweise be copy&pasted all over)
* convert the `Page<*Entity>` to `Page<*Dto>` - as we are forced to use MapStruct so far we can not solve this in a generic way without lambdas was seems a little over-engineered.

So this review comment is just a notice and not a task :)
If we do support paging here, then we have to do it properly.
Otherwise we could assume that the number of animals (or products) is strictly limited and we do not need paging at all and would remove it. However, I see that paging for products fully makes sense as you can have a huge catalog.